### DISCUSSION TITLE
Export realm users dont work Keycloak 18 (Docker)

### DISCUSSION THREAD

**Discussion by Hermansson** [keycloak/keycloak#13405] [Q&A]
Hi.

I'm trying to export users by:

- docker exec -it d17b098... bash
- /opt/keycloak/bin/kc.sh export --dir /opt/keycloak/data/import --realm Demo --users realm_file

the console informs me:
- "Exporting into directory /opt/keycloak/data/import"
- "Exporting into directory /opt/keycloak/data/import"
- "Export of realm 'Demo' requested."
- "Failed to start server in (import_export) mode"

The "/opt/keycloak/data/import" remains empty.

Am I missing something? I'n running Keycloak 18.0.2 in dev mode. Exportering the realm via the admin console works fine, although I would like my users with roles and groups included.

Importing an existing realm by using the "--import-realm" option works fine. I did that, added some users, and is now trying to export it again.

I've been following this tutorial: https://www.helikube.de/keycloak-18-export-and-import-feature/

Grateful for any input.

**Comment by lecranek** [comment]
Same issue. I run my Keycloak instance on Kubernetes using helm chart keycloak:9.7.1 from https://charts.bitnami.com/bitnami, which is based on Keycloak:18.0.2. When executing command `bin/kc.sh --verbose export --dir /tmp/json -
-users realm_file --realm app`, the below error will be thrown:
```bash
2022-09-02 03:02:29,724 INFO  [org.keycloak.services] (main) KC-SERVICES0035: Export finis
hed successfully
2022-09-02 03:02:29,993 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler]
 (main) ERROR: Failed to start server in (import_export) mode
2022-09-02 03:02:29,994 ERROR [org.keycloak.quarkus.runtime.cli.ExecutionExceptionHandler]
 (main) Error details:: java.lang.RuntimeException: Unable to start HTTP server
        at io.quarkus.vertx.http.runtime.VertxHttpRecorder.doServerStart(VertxHttpRecorder
.java:624)
...
```

**Comment by axeh** [comment]
Same issue.  I'm running Keycloak 19.0.3 in dev mode

**Comment by Hermansson** [comment]
Have not tried this yet. But sounds like json-fil should be edited before running the import, removing som policies? Although, I was not trying to import from an old Keycloak version.

https://www.helikube.de/keycloak-18-export-and-import-feature/

**Comment by GianniAzizi** [comment]
You encounter this problem probably because your Keycloak instance is already running. 
According to the documentation, your instance must not be started to run export command : 

[https://www.keycloak.org/server/importExport](https://www.keycloak.org/server/importExport)

I found a workaround when you use docker which is to start Keycloak on another port then run export command that run it on port 8080 which is not used.

To run on another port use this option : 

`start-dev --http-port=8180`
